I am testing a python claculation

In [1]:
1+1

2

In [2]:
100/6

16.666666666666668

In [3]:
77/9

8.555555555555555

In [5]:
pi=3.14
pi*3

9.42

In [6]:
import pandas as pd

# URL of the Titanic dataset
url = "https://web.stanford.edu/class/archive/cs/cs109/cs109.1166/stuff/titanic.csv"

# Read the dataset into a pandas DataFrame
titanic_data = pd.read_csv(url)

# Display the first few rows of the dataset
titanic_data.head()


Survived  Pclass                                               Name  \
0         0       3                             Mr. Owen Harris Braund   
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   
3         1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4         0       3                            Mr. William Henry Allen   

      Sex   Age  Siblings/Spouses Aboard  Parents/Children Aboard     Fare  
0    male  22.0                        1                        0   7.2500  
1  female  38.0                        1                        0  71.2833  
2  female  26.0                        0                        0   7.9250  
3  female  35.0                        1                        0  53.1000  
4    male  35.0                        0                        0   8.0500

This is the (a) "amusing, funny, or otherwise interesting dataset" which (b) has missing values and is (c) available online through a URL link to a csv file that I accessed from ChatGPT. It was a big struggle since the links and url Chat provided do not work. I finally use Python directly in Jupyter Notebook to download this dataset.

In [11]:
titanic_data.describe()

Survived      Pclass         Age  Siblings/Spouses Aboard  \
count  887.000000  887.000000  887.000000               887.000000   
mean     0.385569    2.305524   29.471443                 0.525366   
std      0.487004    0.836662   14.121908                 1.104669   
min      0.000000    1.000000    0.420000                 0.000000   
25%      0.000000    2.000000   20.250000                 0.000000   
50%      0.000000    3.000000   28.000000                 0.000000   
75%      1.000000    3.000000   38.000000                 1.000000   
max      1.000000    3.000000   80.000000                 8.000000   

       Parents/Children Aboard       Fare  
count               887.000000  887.00000  
mean                  0.383315   32.30542  
std                   0.807466   49.78204  
min                   0.000000    0.00000  
25%                   0.000000    7.92500  
50%                   0.000000   14.45420  
75%                   0.000000   31.13750  
max                   6.000000  512.32920

I was looking for missing data but I got the summary of quantitative variables

In [9]:
titanic_data['Sex'].value_counts(dropna=False)

Sex
male      573
female    314
Name: count, dtype: int64

In [16]:
import pandas as pd
print(pd.isna(titanic_data))


     Survived  Pclass   Name    Sex    Age  Siblings/Spouses Aboard  \
0       False   False  False  False  False                    False   
1       False   False  False  False  False                    False   
2       False   False  False  False  False                    False   
3       False   False  False  False  False                    False   
4       False   False  False  False  False                    False   
..        ...     ...    ...    ...    ...                      ...   
882     False   False  False  False  False                    False   
883     False   False  False  False  False                    False   
884     False   False  False  False  False                    False   
885     False   False  False  False  False                    False   
886     False   False  False  False  False                    False   

     Parents/Children Aboard   Fare  
0                      False  False  
1                      False  False  
2                      False  Fal

The first data set is not missing values, so I asked chatgpt to provide me with another one.

In [3]:
import pandas as pd

# Path to the extracted CSV file
file_path = "AirQualityUCI.csv"

# Read the dataset into a pandas DataFrame
air_quality_data = pd.read_csv(file_path, sep=';')

# Display the first few rows of the dataset
air_quality_data.head()


Date      Time CO(GT)  PT08.S1(CO)  NMHC(GT) C6H6(GT)  PT08.S2(NMHC)  \
0  10/03/2004  18.00.00    2,6       1360.0     150.0     11,9         1046.0   
1  10/03/2004  19.00.00      2       1292.0     112.0      9,4          955.0   
2  10/03/2004  20.00.00    2,2       1402.0      88.0      9,0          939.0   
3  10/03/2004  21.00.00    2,2       1376.0      80.0      9,2          948.0   
4  10/03/2004  22.00.00    1,6       1272.0      51.0      6,5          836.0   

   NOx(GT)  PT08.S3(NOx)  NO2(GT)  PT08.S4(NO2)  PT08.S5(O3)     T    RH  \
0    166.0        1056.0    113.0        1692.0       1268.0  13,6  48,9   
1    103.0        1174.0     92.0        1559.0        972.0  13,3  47,7   
2    131.0        1140.0    114.0        1555.0       1074.0  11,9  54,0   
3    172.0        1092.0    122.0        1584.0       1203.0  11,0  60,0   
4    131.0        1205.0    116.0        1490.0       1110.0  11,2  59,6   

       AH  Unnamed: 15  Unnamed: 16  
0  0,7578          NaN          NaN  
1  0,7255          NaN          NaN  
2  0,7502          NaN          NaN  
3  0,7867          NaN          NaN  
4  0,7888          NaN          NaN

In [6]:
import pandas as pd
print(pd.isna(air_quality_data))

       Date   Time  CO(GT)  PT08.S1(CO)  NMHC(GT)  C6H6(GT)  PT08.S2(NMHC)  \
0     False  False   False        False     False     False          False   
1     False  False   False        False     False     False          False   
2     False  False   False        False     False     False          False   
3     False  False   False        False     False     False          False   
4     False  False   False        False     False     False          False   
...     ...    ...     ...          ...       ...       ...            ...   
9466   True   True    True         True      True      True           True   
9467   True   True    True         True      True      True           True   
9468   True   True    True         True      True      True           True   
9469   True   True    True         True      True      True           True   
9470   True   True    True         True      True      True           True   

      NOx(GT)  PT08.S3(NOx)  NO2(GT)  PT08.S4(NO2)  PT08.S5(O3)

In [5]:
print(air_quality_data.isnull().sum())  # missing per column

Date              114
Time              114
CO(GT)            114
PT08.S1(CO)       114
NMHC(GT)          114
C6H6(GT)          114
PT08.S2(NMHC)     114
NOx(GT)           114
PT08.S3(NOx)      114
NO2(GT)           114
PT08.S4(NO2)      114
PT08.S5(O3)       114
T                 114
RH                114
AH                114
Unnamed: 15      9471
Unnamed: 16      9471
dtype: int64


After checking, there are 114 data missing in each of 15 columns presented above. And also 2 columns are unnamed, they are missing everything.

Here's a summary of my interaction with ChatGPT:


In this session, I requested assistance in finding a dataset that meets specific criteria for an assignment. The requirements were for a dataset that:

Is amusing, funny, or otherwise interesting.
Contains missing values.
Is available online via a URL link to a CSV file.
I initially suggested a couple of datasets from FiveThirtyEight, but they were not accessible. After several attempts, I provided an alternative dataset, the Air Quality Dataset, which is available for download as a ZIP file from the UCI Machine Learning Repository. This dataset includes missing values in various columns, such as CO levels, and can be extracted and loaded into a Jupyter Notebook for analysis.

And here is the final working verson of the code that we created:

In [ ]:
import pandas as pd

# Path to the extracted CSV file
file_path = "path_to_extracted_folder/AirQualityUCI.csv"  # Update with the actual path

# Read the dataset into a pandas DataFrame
air_quality_data = pd.read_csv(file_path, sep=';')

# Display the first few rows of the dataset
print(air_quality_data.head())

# Check for missing values in the dataset
print(air_quality_data.isnull().sum())


Finally, here's the transcript log from chatgpt.
https://chatgpt.com/share/66f1675c-1c74-8006-8a01-5c516d8002d0


I also included the link to my github account with my work for STA130.
https://github.com/JimmyWooHoo/STA130